cinema/views.py

```diff
@@ -82,12 +104,24 @@
         .select_related("movie", "cinema_hall")
         .annotate(
             tickets_available=F("cinema_hall__rows")
-            * F("cinema_hall__seats_in_row")
-            - Count("tickets")
+                              * F("cinema_hall__seats_in_row")
+                              - Count("tickets")
         )
     )
     serializer_class = MovieSessionSerializer
 
+    def get_permissions(self):
+        if self.action in (
+                "list",
+                "create",
+                "retrieve",
+                "update",
+                "partial_update",
+                "destroy"
+        ):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
```

Comment:
In `MovieSessionViewSet`, the `get_permissions` method allows all actions for authenticated users. Double-check if actions like 'create', 'update', and 'destroy' should be restricted to admins.